lib/index.ts

```diff
@@ -29,13 +29,26 @@ export const generateRemoteUrl = (service: string) => window.location.protocol +
 /**
  * Get the base path for the given OCS API service
  *
- * @param {string} service name
  * @param {int} version OCS API version
- * @return {string} OCS API base path
+ * @param {string} url OCS API service url
+ * @param {object} params parameters to be replaced into the service url
+ * @param {UrlOptions} options options for the parameter replacement
+ * @return {string} Absolute path for the OCS URL
  */
-export const generateOcsUrl = (service: string, version: Number) => {
+export const generateOcsUrl = (version, url, params?: object, options?: UrlOptions) => {
+    const legacyMode = typeof version === 'string'
```

Comment:
I now introduced a legacy mode in the function, not sure if that makes things better or worse, but it's now fully backwards compatible.

Reply:
While doing so I wondered if the ocs-version should be moved to `UrlOptions`. That one has already a `noRewrite` so we could also have ocsVersion there?

Reply:
let's just make a clear cut without duck typing

I'm fine with moving the version to the options, then it can also have a proper default of `2`

Reply:
okay, done